crates/uv-settings/src/lib.rs

```diff
@@ -303,6 +303,9 @@ fn warn_uv_toml_masked_fields(options: &Options) {
                 concurrent_builds,
                 concurrent_installs,
                 allow_insecure_host,
+                http_proxy,
+                https_proxy,
+                no_proxy,
```

Comment:
I started looking at this and I feel like we might need a `validate_pyproject_toml` function like `validate_uv_toml`? We won't hard deny any of these options in the `pyproject.toml` toda.

I'm surprised that we allow `allow_insecure_host` in the `pyproject.toml` today, which might mean my and @samypr100's points aren't really in scope for this work.

Reply:
I think my understanding of what we weren't allowing today is wrong and it probably doesn't make sense to block this on fixing that.